plugins/content/adaptiveimage/adaptiveimage.php

```diff
@@ -110,4 +111,18 @@ class = \"adaptiveimg\"/>";
 
 		return true;
 	}
+	/**
+	 * Call to the method of respective	class for the respective object passed
+	 * 
+	 * @param   FocusStoreInterface  $storage  Storage Object
+	 * @param   String               $imgPath  Image Path
+	 * 
+	 * @return  Array
+	 * 
+	 * @since 4.0.0
+	 */
+	protected function performTask(FocusStoreInterface $storage, $imgPath)
+	{
+		return $storage->getFocus($imgPath);
+	}
```

Comment:
What's the benefit of a separate method here (and above)?

Reply:
As Discussed, this plugin would not call the controller, rather it will call Interface directly.

Reply:
there is no need to have this performTask method in the plugin as @nibra stated

Reply:
what you can do this is instantiate a storage class in the plugin, pass it over to controller to take the action it wants